### PR TITLE
meta: VS Code: add "Git: Always Sign Off" setting

### PR DESCRIPTION
**Pull request by bam80**
VS Code setting to auto signoff commits.
This also corresponds to our GitHub Web workflow.

Signed-off-by: Andrey Butirsky <butirsky@gmail.com>

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "git.alwaysSignOff": true
+}
```
Comment (neheb) [inline .vscode/settings.json:3]: newline please.

Comment (bam80) [reply .vscode/settings.json]: done

**Review by neheb — APPROVED** [review]

**Comment by neheb** [comment]
Actually isn't there an always newline setting in VSCode too?

**Comment by neheb** [comment]
Ah it's actually editorconfig.

**Comment by bam80** [comment]
@neheb so you can't actually merge this, and we are waiting for someone who can?

**Comment by neheb** [comment]
No

**Comment by bam80** [comment]
> Ah it's actually editorconfig.

@neheb Hi, could you clarify the setting for a new line, and do you think we should apply it, too?

**Comment by neheb** [comment]
```
~/.editorconfig
[*]
charset = utf-8
insert_final_newline = true
indent_style = tab
trim_trailing_whitespace=true
```

probably.

**Comment by bam80** [comment]
Hmm, I also found we have this in the settings.json:
```json
"files.eol": "\n"
```
https://github.com/openwrt/packages/blob/81b237d43a0a20ae125f1edfbad60372a509bf4a/.vscode/settings.json#L6

**Comment by neheb** [comment]
ping @BKPepe

**Comment by BKPepe** [comment]
Yeah, we should sync those changes from packages repo. I will look into it and send PR.

**Comment by bam80** [comment]
> `~/.editorconfig`

The .editorconfig seems more portable, just need to place it in this repo, instead of $HOME.
That would require an extension for VS Code, though.
So maybe have both.

**Pull request merged**